lib/rubocop/cop/util.rb

```diff
@@ -148,6 +148,10 @@ def move_pos(src, pos, step, condition, regexp)
         pos
       end
 
+      def looking_at(range, regexp)
+        range.source_buffer.source.match(/\G#{regexp}/, range.begin_pos)
```

Comment:
@jonas054 If you omit `\G`, then it will stop registering an offence for something like `proc{ -> {} }` or even `proc{ "->" }`.  By `\G` `looking_at(/(?<=->)/)` should properly check whether the arrow `->` is placed right behind the range beginning.